backend/src/test/java/codezap/template/service/TemplateServiceTest.java

```diff
@@ -7,272 +7,566 @@
+                    () -> assertThat(actual.getContent()).hasSize(2),
+                    () -> assertThat(actual.getContent())
+                            .allMatch(template -> template.getCategory().getId().equals(category1.getId()))
+            );
+        }
+
+        @Test
+        @DisplayName("검색 기능: 태그 ID 목록으로 템플릿 조회, 모든 태그를 가진 템플릿만 조회 성공")
+        void findAllSuccessByTagIds() {
+            saveInitialData();
+            Long memberId = null;
+            String keyword = null;
+            Long categoryId = null;
+            List<Long> tagIds = List.of(tag1.getId(), tag2.getId());
+            Pageable pageable = PageRequest.of(0, 10);
+
+            Page<Template> actual = sut.findAll(memberId, keyword, categoryId, tagIds, pageable);
+
+            assertAll(
+                    () -> assertThat(actual.getContent())
+                            .containsExactlyInAnyOrder(
+                                    templateRepository.fetchById(1L),
+                                    templateRepository.fetchById(2L)),
+                    () -> assertThat(actual.getContent()).hasSize(2)
+            );
+        }
+
+        @Test
+        @DisplayName("검색 기능: 단일 태그 ID로 템플릿 조회 성공")
+        void findAllSuccessBySingleTagId() {
+            saveInitialData();
+            Long memberId = null;
+            String keyword = null;
+            Long categoryId = null;
+            List<Long> tagIds = List.of(tag2.getId());
+            Pageable pageable = PageRequest.of(0, 10);
+
+            Page<Template> actual = sut.findAll(memberId, keyword, categoryId, tagIds, pageable);
+
+            assertAll(
+                    () -> assertThat(actual.getContent()).containsExactlyInAnyOrder(
+                            templateRepository.fetchById(1L),
+                            templateRepository.fetchById(2L),
+                            templateRepository.fetchById(3L)),
+                    () -> assertThat(actual.getContent()).hasSize(3)
+            );
+        }
+
+        @Test
+        @DisplayName("검색 기능: 회원 ID와 키워드로 템플릿 조회 성공")
+        void findAllSuccessByMemberIdAndKeyword() {
+            saveInitialData();
+            Long memberId = member1.getId();
+            String keyword = "Template";
+            Long categoryId = null;
+            List<Long> tagIds = null;
+            Pageable pageable = PageRequest.of(0, 10);
+
+            Page<Template> actual = sut.findAll(memberId, keyword, categoryId, tagIds, pageable);
+
+            assertAll(
+                    () -> assertThat(actual.getContent()).hasSize(2),
+                    () -> assertThat(actual.getContent())
+                            .allMatch(template ->
+                                    template.getMember().getId().equals(member1.getId())
+                                            && (template.getTitle().contains(keyword) || template.getDescription()
+                                            .contains(keyword)))
+            );
```

Comment:
```suggestion
                                    template.getMember().getId().equals(member1.getId())
                                            && (template.getTitle().contains(keyword) 
                                            || template.getDescription().contains(keyword)))
            );
```
으엥 이렇게 보니까 이상한 것 같긴 한데 .. 암튼 요런 느낌.. 어떨까요?
<img width="414" alt="image" src="https://github.com/user-attachments/assets/898e9159-21dc-44cc-ae08-2f3840743ba7">

Reply:
저도 몰리 의견에 한 표 추가합니다!

[6.4. 줄바꿈 허용 위치](https://naver.github.io/hackday-conventions-java/#line-wrapping-position)
네이버 핵데이 컨벤션인데요. 여기서도 연산자 앞에서 줄바꿈을 규칙으로 지정하고 있어요.

조건 연산자로 연결된 줄이 길어질 경우, 조건 연산자 앞에서 개행을 해주면 훨씬 가독성이 좋아지는 것 같네용

Reply:
해당 테스트 코드들을 `TemplateJpaRepositoryTest`에서 가져와서..
줄바꿈 수정하는 김에 해당 클래스도 같이 수정해서 올리겠습니다.